core/components/batchedWriterAssembly.go

```diff
@@ -109,3 +110,55 @@ func (bwa *BatchedWriterAssembly) FlushOnTimeOut() {
 		bwa.Flush()
 	}
 }
+
+// NeedsRotate evaluate if the BatchedWriterAssembly need to rotate by the FileRotateConfig
+func (bwa *BatchedWriterAssembly) NeedsRotate(rotate RotateConfig, forceRotate bool) (bool, error) {
+	// File does not exist?
+	if !bwa.HasWriter() {
+		bwa.logger.Debug("Rotate true: ", "no writer")
+		return true, nil
+	}
+
+	// File can be accessed?
+	if bwa.GetWriter().IsAccessible() == false {
+		bwa.logger.Debug("Rotate false: ", "no access")
+		return false, errors.New("Can' access file to rotate")
+	}
+
+	// File needs rotation?
+	if !rotate.Enabled {
+		bwa.logger.Debug("Rotate false: ", "not active")
+		return false, nil
+	}
+
+	if forceRotate {
+		bwa.logger.Debug("Rotate true: ", "forced")
+		return true, nil
+	}
+
+	// File is too large?
+	if bwa.GetWriter().Size() >= rotate.SizeByte {
+		bwa.logger.Debug("Rotate true: ", "size > rotation size")
+		return true, nil // ### return, too large ###
+	}
+
+	// File is too old?
+	if time.Since(bwa.Created) >= rotate.Timeout {
+		bwa.logger.Debug("Rotate true: ", "lifetime > timeout setting")
+		return true, nil // ### return, too old ###
+	}
+
+	// RotateAt crossed?
+	if rotate.AtHour > -1 && rotate.AtMinute > -1 {
+		now := time.Now()
+		rotateAt := time.Date(now.Year(), now.Month(), now.Day(), rotate.AtHour, rotate.AtMinute, 0, 0, now.Location())
+
```

Comment:
Example:
- rotateAt is set to 11:00
- the file was created at 11:01
- we hit this code at  11:06

As rotateAt lies before created we need to move rotateAt 1 day into the future (don't rotate).
Now if we pass midnight, rotateAt will be after created (different day) and this check will not pass, i.e. no day is added. When we now pass the mark, a rotation will be triggered.